From 6b2794279012ce68bf59ab56ea49cca24d724f6e Mon Sep 17 00:00:00 2001
From: Suneel Garapati <sgarapati@marvell.com>
Date: Fri, 8 Nov 2019 17:16:59 -0800
Subject: [PATCH 0992/1239] octeontx: octeontx: code cleanup after rebase from
 v2018.03

This patch cleans up all redundant code changes and also
which are irrelevant to v2019.10. Move files to correct location
as necessary like nand driver. Some cleanup wrt checkpatch too.

Change-Id: I7ae2c015d67f8aed25f3fe3a694c2d1cb1702105
Signed-off-by: Suneel Garapati <sgarapati@marvell.com>
---
 arch/arm/Kconfig                              |   9 +-
 arch/arm/Makefile                             |   2 -
 arch/arm/include/asm/arch-octeontx/board.h    |  74 ---------
 arch/arm/include/asm/arch-octeontx2/board.h   |  52 ------
 arch/arm/include/asm/gpio.h                   |   2 +-
 arch/arm/include/asm/io.h                     |   1 -
 arch/arm/include/asm/system.h                 |  35 -----
 arch/arm/lib/bootm.c                          |   4 -
 arch/arm/lib/interrupts_64.c                  |   5 -
 board/Marvell/octeontx/board-fdt.c            |   2 +-
 board/Marvell/octeontx/board.c                |  12 --
 board/Marvell/octeontx2/board-fdt.c           |   2 +-
 board/Marvell/octeontx2/board.c               |  16 +-
 cmd/Kconfig                                   |  10 --
 cmd/bootm.c                                   |   1 -
 cmd/nvme.c                                    |   1 +
 cmd/scsi.c                                    |  16 --
 common/board_r.c                              |   5 +-
 common/ddr_spd.c                              |  34 ----
 configs/t88_defconfig                         |  50 ------
 drivers/ata/ahci.c                            |  12 +-
 drivers/block/Makefile                        |   3 -
 drivers/block/blk-uclass.c                    |   1 +
 drivers/i2c/Kconfig                           |  11 +-
 drivers/i2c/octeontx_i2c.c                    |   6 +
 drivers/mmc/Kconfig                           |   8 +-
 drivers/mmc/Makefile                          |   2 +-
 drivers/mmc/mmc-uclass.c                      |  14 +-
 drivers/mmc/octeontx_hsmmc.c                  |  46 +-----
 drivers/mmc/octeontx_hsmmc.h                  |   2 +
 drivers/mtd/nand/Kconfig                      |   1 -
 drivers/mtd/nand/raw/Kconfig                  |  31 ++--
 drivers/mtd/nand/{ => raw}/octeontx_bch.c     |   0
 drivers/mtd/nand/{ => raw}/octeontx_bch.h     |   0
 .../mtd/nand/{ => raw}/octeontx_bch_regs.h    |   0
 drivers/mtd/nand/{ => raw}/octeontx_nand.c    |  14 --
 drivers/net/e1000.c                           |   1 -
 drivers/net/e1000_spi.c                       |   4 +-
 drivers/net/octeontx/nic_main.c               |   2 -
 drivers/net/octeontx/nicvf_main.c             |   2 -
 drivers/net/octeontx/smi.c                    |   2 +-
 drivers/net/octeontx2/nix_af.c                |   4 +-
 drivers/pci/pci_auto.c                        |   2 -
 drivers/spi/spi-uclass.c                      |  33 ----
 drivers/spi/spi.c                             |   1 -
 env/mmc.c                                     |  17 +-
 env/sf.c                                      |   2 +-
 include/common.h                              |   1 -
 include/ddr_spd.h                             |   2 -
 include/mmc.h                                 |  30 +---
 include/pci.h                                 |  48 +-----
 include/pci_ids.h                             |   5 +-
 include/phy.h                                 |  54 -------
 include/spi.h                                 | 148 ------------------
 net/Kconfig                                   |  28 ----
 net/eth_legacy.c                              |   3 -
 56 files changed, 85 insertions(+), 788 deletions(-)
 delete mode 100644 configs/t88_defconfig
 rename drivers/mtd/nand/{ => raw}/octeontx_bch.c (100%)
 rename drivers/mtd/nand/{ => raw}/octeontx_bch.h (100%)
 rename drivers/mtd/nand/{ => raw}/octeontx_bch_regs.h (100%)
 rename drivers/mtd/nand/{ => raw}/octeontx_nand.c (99%)

diff --git a/arch/arm/Kconfig b/arch/arm/Kconfig
index 97407cab86..49b2061de2 100644
--- a/arch/arm/Kconfig
+++ b/arch/arm/Kconfig
@@ -390,6 +390,7 @@ config TPL_SYS_THUMB_BUILD
 	   density. For ARM architectures that support Thumb2 this flag will
 	   result in Thumb2 code generated by GCC.
 
+
 config SYS_L2CACHE_OFF
 	bool "L2cache off"
 	help
@@ -464,7 +465,7 @@ config TPL_USE_ARCH_MEMSET
 config ARM64_SUPPORT_AARCH32
 	bool "ARM64 system support AArch32 execution state"
 	depends on ARM64
-	default y if !TARGET_OCTEONTX_88XX
+	default y if !TARGET_THUNDERX_88XX
 	help
 	  This ARM64 system supports AArch32 execution state.
 
@@ -1614,8 +1615,8 @@ config ARCH_OCTEONTX2
 	select BOARD_LATE_INIT
 	select SYS_CACHE_SHIFT_7
 
-config TARGET_OCTEONTX_88XX
-	bool "Support OcteonTX 88xx"
+config TARGET_THUNDERX_88XX
+	bool "Support ThunderX 88xx"
 	select ARM64
 	select OF_CONTROL
 	select PL01X_SERIAL
@@ -1701,8 +1702,6 @@ source "arch/arm/mach-imx/mx2/Kconfig"
 
 source "arch/arm/mach-imx/mx3/Kconfig"
 
-source "arch/arm/mach-thunderx/Kconfig"
-
 source "arch/arm/mach-imx/mx5/Kconfig"
 
 source "arch/arm/mach-imx/mx6/Kconfig"
diff --git a/arch/arm/Makefile b/arch/arm/Makefile
index 9d8c36347b..80c7c61061 100644
--- a/arch/arm/Makefile
+++ b/arch/arm/Makefile
@@ -43,8 +43,6 @@ tune-$(CONFIG_CPU_ARM1176)	=
 tune-$(CONFIG_CPU_V7A)		=-mtune=generic-armv7-a
 tune-$(CONFIG_CPU_V7R)		=
 tune-$(CONFIG_ARM64)		=
-tune-$(CONFIG_OCTEONTX)		=
-tune-$(CONFIG_OCTEONTX2)	=
 
 # Evaluate tune cc-option calls now
 tune-y := $(tune-y)
diff --git a/arch/arm/include/asm/arch-octeontx/board.h b/arch/arm/include/asm/arch-octeontx/board.h
index 2ed454d1c8..8755226a31 100644
--- a/arch/arm/include/asm/arch-octeontx/board.h
+++ b/arch/arm/include/asm/arch-octeontx/board.h
@@ -29,9 +29,6 @@
 /** Reg offsets */
 #define RST_BOOT	0x87E006001600ULL
 
-#define GTI_CWD_POKEX	0x844000050000ULL
-#define GTI_CWD_WDOGX	0x844000040000ULL
-
 /** Structure definitions */
 
 /**
@@ -109,77 +106,6 @@ union rst_boot {
 		u64 jtcsrdis                         : 1;
 		u64 chipkill                         : 1;
 	} cn83xx;
-	struct rst_boot_cn96xx {
-		u64 rboot_pin                        : 1;
-		u64 rboot                            : 1;
-		u64 reserved_2_23                    : 22;
-		u64 cpt_mul                          : 7;
-		u64 reserved_31_32                   : 2;
-		u64 pnr_mul                          : 6;
-		u64 reserved_39                      : 1;
-		u64 c_mul                            : 7;
-		u64 reserved_47_52                   : 6;
-		u64 gpio_ejtag                       : 1;
-		u64 mcp_jtagdis                      : 1;
-		u64 dis_scan                         : 1;
-		u64 dis_huk                          : 1;
-		u64 vrm_err                          : 1;
-		u64 reserved_58_59                   : 2;
-		u64 trusted_mode                     : 1;
-		u64 scp_jtagdis                      : 1;
-		u64 jtagdis                          : 1;
-		u64 chipkill                         : 1;
-	} cn96xx;
-	struct rst_boot_cn98xx {
-		u64 rboot_pin                        : 1;
-		u64 rboot                            : 1;
-		u64 reserved_2_7                     : 6;
-		u64 rxp_mul                          : 7;
-		u64 reserved_15                      : 1;
-		u64 cpt1_mul                         : 7;
-		u64 reserved_23                      : 1;
-		u64 cpt_mul                          : 7;
-		u64 reserved_31_32                   : 2;
-		u64 pnr_mul                          : 6;
-		u64 reserved_39                      : 1;
-		u64 c_mul                            : 7;
-		u64 reserved_47_52                   : 6;
-		u64 gpio_ejtag                       : 1;
-		u64 mcp_jtagdis                      : 1;
-		u64 dis_scan                         : 1;
-		u64 dis_huk                          : 1;
-		u64 vrm_err                          : 1;
-		u64 reserved_58_59                   : 2;
-		u64 trusted_mode                     : 1;
-		u64 scp_jtagdis                      : 1;
-		u64 jtagdis                          : 1;
-		u64 chipkill                         : 1;
-	} cn98xx;
-	struct rst_boot_cnf95xx {
-		u64 rboot_pin                        : 1;
-		u64 rboot                            : 1;
-		u64 reserved_2_7                     : 6;
-		u64 bphy_mul                         : 7;
-		u64 reserved_15                      : 1;
-		u64 dsp_mul                          : 7;
-		u64 reserved_23                      : 1;
-		u64 cpt_mul                          : 7;
-		u64 reserved_31_32                   : 2;
-		u64 pnr_mul                          : 6;
-		u64 reserved_39                      : 1;
-		u64 c_mul                            : 7;
-		u64 reserved_47_52                   : 6;
-		u64 gpio_ejtag                       : 1;
-		u64 mcp_jtagdis                      : 1;
-		u64 dis_scan                         : 1;
-		u64 dis_huk                          : 1;
-		u64 vrm_err                          : 1;
-		u64 reserved_58_59                   : 2;
-		u64 trusted_mode                     : 1;
-		u64 scp_jtagdis                      : 1;
-		u64 jtagdis                          : 1;
-		u64 chipkill                         : 1;
-	} cnf95xx;
 };
 
 /** Function definitions */
diff --git a/arch/arm/include/asm/arch-octeontx2/board.h b/arch/arm/include/asm/arch-octeontx2/board.h
index 92d98e0a0e..6d3691e809 100644
--- a/arch/arm/include/asm/arch-octeontx2/board.h
+++ b/arch/arm/include/asm/arch-octeontx2/board.h
@@ -12,13 +12,9 @@
 
 /** Reg offsets */
 #define RST_BOOT		0x87E006001600ULL
-#define RST_CHIP_DOM_W1S	0x87E006001810ULL
 
 #define CPC_BOOT_OWNERX(a)	0x86D000000160ULL + (8 * (a))
 
-#define GTI_CWD_POKEX	0x802000050000ULL
-#define GTI_CWD_WDOGX	0x802000040000ULL
-
 /* attestation definitions shared with ATF (see 'plat_octeontx.h') */
 
 #define ATTESTATION_MAGIC_ID 0x5f415454 /* "_ATT" */
@@ -109,54 +105,6 @@ union rst_boot {
 		u64 reserved_61_62                   : 2;
 		u64 chipkill                         : 1;
 	} s;
-	struct rst_boot_cn81xx {
-		u64 rboot_pin                        : 1;
-		u64 rboot                            : 1;
-		u64 lboot                            : 10;
-		u64 lboot_ext23                      : 6;
-		u64 lboot_ext45                      : 6;
-		u64 lboot_jtg                        : 1;
-		u64 lboot_ckill                      : 1;
-		u64 reserved_26_29                   : 4;
-		u64 lboot_oci                        : 3;
-		u64 pnr_mul                          : 6;
-		u64 reserved_39                      : 1;
-		u64 c_mul                            : 7;
-		u64 reserved_47_54                   : 8;
-		u64 dis_scan                         : 1;
-		u64 dis_huk                          : 1;
-		u64 vrm_err                          : 1;
-		u64 jt_tstmode                       : 1;
-		u64 ckill_ppdis                      : 1;
-		u64 trusted_mode                     : 1;
-		u64 ejtagdis                         : 1;
-		u64 jtcsrdis                         : 1;
-		u64 chipkill                         : 1;
-	} cn81xx;
-	struct rst_boot_cn83xx {
-		u64 rboot_pin                        : 1;
-		u64 rboot                            : 1;
-		u64 lboot                            : 10;
-		u64 lboot_ext23                      : 6;
-		u64 lboot_ext45                      : 6;
-		u64 lboot_jtg                        : 1;
-		u64 lboot_ckill                      : 1;
-		u64 lboot_pf_flr                     : 4;
-		u64 lboot_oci                        : 3;
-		u64 pnr_mul                          : 6;
-		u64 reserved_39                      : 1;
-		u64 c_mul                            : 7;
-		u64 reserved_47_54                   : 8;
-		u64 dis_scan                         : 1;
-		u64 dis_huk                          : 1;
-		u64 vrm_err                          : 1;
-		u64 jt_tstmode                       : 1;
-		u64 ckill_ppdis                      : 1;
-		u64 trusted_mode                     : 1;
-		u64 ejtagdis                         : 1;
-		u64 jtcsrdis                         : 1;
-		u64 chipkill                         : 1;
-	} cn83xx;
 	struct rst_boot_cn96xx {
 		u64 rboot_pin                        : 1;
 		u64 rboot                            : 1;
diff --git a/arch/arm/include/asm/gpio.h b/arch/arm/include/asm/gpio.h
index 86da0c080b..6ff5f42424 100644
--- a/arch/arm/include/asm/gpio.h
+++ b/arch/arm/include/asm/gpio.h
@@ -3,7 +3,7 @@
 	!defined(CONFIG_ARCH_BCM63158) && !defined(CONFIG_ARCH_ROCKCHIP) && \
 	!defined(CONFIG_ARCH_LX2160A) && !defined(CONFIG_ARCH_LS1028A) && \
 	!defined(CONFIG_ARCH_LS2080A) && !defined(CONFIG_ARCH_LS1088A) && \
-	!defined(CONFIG_ARCH_ASPEED) && !defined(CONFIG_ARM64)
+	!defined(CONFIG_ARCH_ASPEED)
 #include <asm/arch/gpio.h>
 #endif
 #include <asm-generic/gpio.h>
diff --git a/arch/arm/include/asm/io.h b/arch/arm/include/asm/io.h
index ebd9a67019..0f51552018 100644
--- a/arch/arm/include/asm/io.h
+++ b/arch/arm/include/asm/io.h
@@ -234,7 +234,6 @@ static inline void __raw_readsl(unsigned long addr, void *data, int longlen)
 #define setbits_le64(addr, set) setbits(le64, addr, set)
 #define clrsetbits_le64(addr, clear, set) clrsetbits(le64, addr, clear, set)
 
-
 /*
  * Now, pick up the machine-defined IO definitions
  */
diff --git a/arch/arm/include/asm/system.h b/arch/arm/include/asm/system.h
index b2bf856c3b..a1a5e35ef6 100644
--- a/arch/arm/include/asm/system.h
+++ b/arch/arm/include/asm/system.h
@@ -176,41 +176,6 @@ static inline unsigned long read_mpidr(void)
 	return val;
 }
 
-static inline unsigned long read_far(void)
-{
-	unsigned int el;
-	unsigned long val;
-
-	el = current_el();
-	if (el == 1)
-		asm volatile("mrs %0, far_el1" : "=r" (val) : : "cc");
-	else if (el == 2)
-		asm volatile("mrs %0, far_el2" : "=r" (val) : : "cc");
-	else
-		asm volatile("mrs %0, far_el3" : "=r" (val) : : "cc");
-
-	asm volatile("isb");
-
-	return val;
-}
-
-static inline long unsigned int read_spsr(void)
-{
-	unsigned int el, val;
-
-	el = current_el();
-	if (el == 1)
-		asm volatile("mrs %0, spsr_el1" : "=r" (val) : : "cc");
-	else if (el == 2)
-		asm volatile("mrs %0, spsr_el2" : "=r" (val) : : "cc");
-	else
-		asm volatile("mrs %0, spsr_el3" : "=r" (val) : : "cc");
-
-	asm volatile("isb");
-
-	return val;
-}
-
 #define BSP_COREID	0
 
 void __asm_flush_dcache_all(void);
diff --git a/arch/arm/lib/bootm.c b/arch/arm/lib/bootm.c
index 2a8c84cd46..1638f1e81d 100644
--- a/arch/arm/lib/bootm.c
+++ b/arch/arm/lib/bootm.c
@@ -221,10 +221,6 @@ static void do_nonsec_virt_switch(void)
 {
 	smp_kick_all_cpus();
 	dcache_disable();	/* flush cache before swtiching to EL2 */
-	armv8_switch_to_el2();
-#ifdef CONFIG_ARMV8_SWITCH_TO_EL1
-	armv8_switch_to_el1();
-#endif
 }
 #endif
 
diff --git a/arch/arm/lib/interrupts_64.c b/arch/arm/lib/interrupts_64.c
index a89f397f2a..33ec856087 100644
--- a/arch/arm/lib/interrupts_64.c
+++ b/arch/arm/lib/interrupts_64.c
@@ -34,11 +34,6 @@ void show_regs(struct pt_regs *regs)
 {
 	int i;
 
-	printf("MPIDR:     %lx\n", read_mpidr());
-	printf("FAR:       %lx\n", read_far());
-	printf("SPSR:      %lx\n", read_spsr());
-	printf("Rel. offs: %lx\n", gd->reloc_off);
-
 	if (gd->flags & GD_FLG_RELOC)
 		printf("elr: %016lx lr : %016lx (reloc)\n",
 		       regs->elr - gd->reloc_off,
diff --git a/board/Marvell/octeontx/board-fdt.c b/board/Marvell/octeontx/board-fdt.c
index 46bae3944d..0f38ee8cba 100644
--- a/board/Marvell/octeontx/board-fdt.c
+++ b/board/Marvell/octeontx/board-fdt.c
@@ -8,7 +8,7 @@
 #include <common.h>
 #include <malloc.h>
 #include <errno.h>
-#include <environment.h>
+#include <env.h>
 #include <asm/io.h>
 #include <linux/compiler.h>
 #include <linux/libfdt.h>
diff --git a/board/Marvell/octeontx/board.c b/board/Marvell/octeontx/board.c
index a5abb79ad1..10c7d2f90b 100644
--- a/board/Marvell/octeontx/board.c
+++ b/board/Marvell/octeontx/board.c
@@ -9,7 +9,6 @@
 #include <dm.h>
 #include <malloc.h>
 #include <errno.h>
-#include <environment.h>
 #include <netdev.h>
 #include <asm/io.h>
 #include <linux/compiler.h>
@@ -150,14 +149,3 @@ int show_board_info(void)
 	return 0;
 }
 
-#ifdef CONFIG_HW_WATCHDOG
-void hw_watchdog_reset(void)
-{
-	writeq(~0ULL, GTI_CWD_POKEX);
-}
-
-void hw_watchdog_disable(void)
-{
-	writeq(0ULL, GTI_CWD_WDOGX);
-}
-#endif
diff --git a/board/Marvell/octeontx2/board-fdt.c b/board/Marvell/octeontx2/board-fdt.c
index 1151cc31ed..1f0ddca08c 100644
--- a/board/Marvell/octeontx2/board-fdt.c
+++ b/board/Marvell/octeontx2/board-fdt.c
@@ -8,7 +8,7 @@
 #include <common.h>
 #include <malloc.h>
 #include <errno.h>
-#include <environment.h>
+#include <env.h>
 #include <asm/io.h>
 #include <linux/compiler.h>
 #include <linux/libfdt.h>
diff --git a/board/Marvell/octeontx2/board.c b/board/Marvell/octeontx2/board.c
index 54e023d0d7..1bb8a2f5c0 100644
--- a/board/Marvell/octeontx2/board.c
+++ b/board/Marvell/octeontx2/board.c
@@ -1,7 +1,8 @@
+// SPDX-License-Identifier:    GPL-2.0
 /*
  * Copyright (C) 2018 Marvell International Ltd.
  *
- * SPDX-License-Identifier:    GPL-2.0
+ * https://spdx.org/licenses
  */
 
 #include <common.h>
@@ -10,7 +11,6 @@
 #include <dm/uclass-internal.h>
 #include <malloc.h>
 #include <errno.h>
-#include <environment.h>
 #include <asm/io.h>
 #include <linux/compiler.h>
 #include <linux/libfdt.h>
@@ -401,18 +401,6 @@ void acquire_flash_arb(bool acquire)
 	}
 }
 
-#ifdef CONFIG_HW_WATCHDOG
-void hw_watchdog_reset(void)
-{
-	writeq(~0ULL, GTI_CWD_POKEX);
-}
-
-void hw_watchdog_disable(void)
-{
-	writeq(0ULL, GTI_CWD_WDOGX);
-}
-#endif
-
 #ifdef CONFIG_LAST_STAGE_INIT
 int last_stage_init(void)
 {
diff --git a/cmd/Kconfig b/cmd/Kconfig
index afb1d079cd..08c1d33ae3 100644
--- a/cmd/Kconfig
+++ b/cmd/Kconfig
@@ -1149,15 +1149,6 @@ config CMD_SATA
 	  Attachment, where AT refers to an IBM AT (Advanced Technology)
 	  computer released in 1984.
 
-config CMD_NVME
-	bool "NVMe"
-	depends on NVME
-	help
-	  NVMe support.
-	  This enables nvme commands in command line
-	  You can use comamnd "nvme" to show what commands it supports
-	  such as "nvme info" "nvme list".
-
 config CMD_SAVES
 	bool "saves - Save a file over serial in S-Record format"
 	help
@@ -1275,7 +1266,6 @@ config CMD_AXI
 	  Interface) busses, a on-chip interconnect specification for managing
 	  functional blocks in SoC designs, which is also often used in designs
 	  involving FPGAs (e.g.  communication with IP cores in Xilinx FPGAs).
-
 endmenu
 
 
diff --git a/cmd/bootm.c b/cmd/bootm.c
index ba947a4ebc..8279f2b7cc 100644
--- a/cmd/bootm.c
+++ b/cmd/bootm.c
@@ -18,7 +18,6 @@
 #include <asm/byteorder.h>
 #include <linux/ctype.h>
 #include <linux/err.h>
-#include <linux/sizes.h>
 #include <u-boot/zlib.h>
 
 DECLARE_GLOBAL_DATA_PTR;
diff --git a/cmd/nvme.c b/cmd/nvme.c
index af345e5522..5da903bd52 100644
--- a/cmd/nvme.c
+++ b/cmd/nvme.c
@@ -20,6 +20,7 @@ static int do_nvme(cmd_tbl_t *cmdtp, int flag, int argc, char * const argv[])
 			ret = nvme_scan_namespace();
 			if (ret)
 				return CMD_RET_FAILURE;
+
 			return ret;
 		}
 		if (strncmp(argv[1], "deta", 4) == 0) {
diff --git a/cmd/scsi.c b/cmd/scsi.c
index 2052b7f50d..67de1a3bdc 100644
--- a/cmd/scsi.c
+++ b/cmd/scsi.c
@@ -64,22 +64,6 @@ U_BOOT_CMD(
 	"     `blk#' from memory address `addr'"
 );
 
-#if !defined(CONFIG_CMD_SATA) && defined(CONFIG_SCSI_AHCI)
-U_BOOT_CMD(
-	sata, 5, 1, do_scsi,
-	"SATA sub-system",
-	"reset - reset SATA controller\n"
-	"sata info  - show available SATA devices\n"
-	"sata init  - (re-)scan SATA bus\n"
-	"sata device [dev] - show or set current device\n"
-	"sata part [dev] - print partition table of one or all SATA devices\n"
-	"sata read addr blk# cnt - read `cnt' blocks starting at block `blk#'\n"
-	"     to memory address `addr'\n"
-	"sata write addr blk# cnt - write `cnt' blocks starting at block\n"
-	"     `blk#' from memory address `addr'"
-);
-#endif
-
 U_BOOT_CMD(
 	scsiboot, 3, 1, do_scsiboot,
 	"boot from SCSI device",
diff --git a/common/board_r.c b/common/board_r.c
index a99e3c5fe5..ce8b12c071 100644
--- a/common/board_r.c
+++ b/common/board_r.c
@@ -415,7 +415,7 @@ static int initr_onenand(void)
 }
 #endif
 
-#if defined(CONFIG_MMC) || defined(CONFIG_DM_MMC)
+#ifdef CONFIG_MMC
 static int initr_mmc(void)
 {
 	puts("MMC:   ");
@@ -457,6 +457,7 @@ static int initr_env(void)
 	env_set_hex("fdtcontroladdr",
 		    (unsigned long)map_to_sysmem(gd->fdt_blob));
 #endif
+
 	/* Initialize from environment */
 	load_addr = env_get_ulong("loadaddr", 16, load_addr);
 
@@ -749,7 +750,7 @@ static init_fnc_t init_sequence_r[] = {
 #ifdef CONFIG_CMD_ONENAND
 	initr_onenand,
 #endif
-#if defined(CONFIG_MMC) || defined(CONFIG_DM_MMC)
+#ifdef CONFIG_MMC
 	initr_mmc,
 #endif
 	initr_env,
diff --git a/common/ddr_spd.c b/common/ddr_spd.c
index da63a1884d..58dc9b3781 100644
--- a/common/ddr_spd.c
+++ b/common/ddr_spd.c
@@ -82,40 +82,6 @@ crc16(char *ptr, int count)
 	return crc & 0xffff;
 }
 
-const char *ddr3_modtype_str(const ddr3_spd_eeprom_t *spd)
-{
-	switch(spd->module_type) {
-	case DDR3_SPD_MODULETYPE_RDIMM:
-		return "Registered DIMM";
-	case DDR3_SPD_MODULETYPE_UDIMM:
-		return "Unbuffered DIMM";
-	case DDR3_SPD_MODULETYPE_SO_DIMM:
-		return "SODIMM";
-	case DDR3_SPD_MODULETYPE_MICRO_DIMM:
-		return "Micro DIMM";
-	case DDR3_SPD_MODULETYPE_MINI_RDIMM:
-		return "Mini RDIMM";
-	case DDR3_SPD_MODULETYPE_MINI_UDIMM:
-		return "Mini UDIMM";
-	case DDR3_SPD_MODULETYPE_MINI_CDIMM:
-		return "Mini CDIMM";
-	case DDR3_SPD_MODULETYPE_72B_SO_UDIMM:
-		return "ECC Unbuffered SODIMM";
-	case DDR3_SPD_MODULETYPE_72B_SO_RDIMM:
-		return "ECC Registered SODIMM";
-	case DDR3_SPD_MODULETYPE_72B_SO_CDIMM:
-		return "ECC CDIMM";
-	case DDR3_SPD_MODULETYPE_LRDIMM:
-		return "LRDIMM";
-	case DDR3_SPD_MODULETYPE_16B_SO_DIMM:
-		return "16B SODIMM";
-	case DDR3_SPD_MODULETYPE_32B_SO_DIMM:
-		return "32B SODIMM";
-	}
-
-	return "Unknown module";
-}
-
 unsigned int
 ddr3_spd_check(const ddr3_spd_eeprom_t *spd)
 {
diff --git a/configs/t88_defconfig b/configs/t88_defconfig
deleted file mode 100644
index 6764bd6813..0000000000
--- a/configs/t88_defconfig
+++ /dev/null
@@ -1,50 +0,0 @@
-CONFIG_ARM=y
-CONFIG_ARCH_THUNDERX=y
-CONFIG_TARGET_THUNDERX_88XX=y
-CONFIG_SYS_TEXT_BASE=0x00500000
-CONFIG_NR_DRAM_BANKS=1
-CONFIG_DEBUG_UART_BASE=0x87e024000000
-CONFIG_DEBUG_UART_CLOCK=24000000
-CONFIG_IDENT_STRING=" for Cavium Thunder CN88XX ARM v8 Multi-Core"
-CONFIG_DEBUG_UART=y
-CONFIG_BOOTDELAY=5
-CONFIG_USE_BOOTARGS=y
-CONFIG_BOOTARGS="console=ttyAMA0,115200n8 earlycon=pl011,0x87e024000000 debug maxcpus=48 rootwait rw root=/dev/sda2 coherent_pool=16M"
-# CONFIG_DISPLAY_CPUINFO is not set
-# CONFIG_DISPLAY_BOARDINFO is not set
-CONFIG_DEFAULT_DEVICE_TREE="thunder-88xx"
-CONFIG_SYS_EXTRA_OPTIONS="ARM64"
-CONFIG_HUSH_PARSER=y
-# CONFIG_AUTO_COMPLETE is not set
-CONFIG_SYS_PROMPT="ThunderX_88XX> "
-# CONFIG_CMD_EXPORTENV is not set
-# CONFIG_CMD_IMPORTENV is not set
-# CONFIG_CMD_EDITENV is not set
-# CONFIG_CMD_SAVEENV is not set
-# CONFIG_CMD_ENV_EXISTS is not set
-# CONFIG_CMD_FLASH is not set
-# CONFIG_CMD_NET is not set
-CONFIG_DEFAULT_DEVICE_TREE="thunderx-88xx"
-# CONFIG_CMD_NFS is not set
-CONFIG_CMD_MEMTEST=y
-CONFIG_CMD_MX_CYCLIC=y
-# CONFIG_CMD_FLASH is not set
-CONFIG_CMD_I2C=y
-CONFIG_CMD_USB=y
-# CONFIG_CMD_FPGA is not set
-CONFIG_CMD_TFTPPUT=y
-CONFIG_CMD_TFTPSRV=y
-CONFIG_CMD_RARP=y
-CONFIG_CMD_DHCP=y
-CONFIG_CMD_PING=y
-CONFIG_CMD_CDP=y
-CONFIG_CMD_SNTP=y
-CONFIG_CMD_DNS=y
-CONFIG_CMD_LINK_LOCAL=y
-CONFIG_NET_RANDOM_ETHADDR=y
-CONFIG_PCI_OCTEONTX_ECAM=y
-CONFIG_DEBUG_UART=y
-CONFIG_DEBUG_UART_PL011=y
-CONFIG_DEBUG_UART_SKIP_INIT=y
-CONFIG_SYS_I2C_SPEED=100000
-CONFIG_SYS_I2C_SLAVE=0x77
diff --git a/drivers/ata/ahci.c b/drivers/ata/ahci.c
index 90c394022d..bc31a715c9 100644
--- a/drivers/ata/ahci.c
+++ b/drivers/ata/ahci.c
@@ -1211,9 +1211,6 @@ int ahci_probe_scsi(struct udevice *ahci_dev, ulong base)
 	ret = ahci_init_one(uc_priv, dev);
 	if (ret)
 		return ret;
-
-	uc_plat->max_id = uc_priv->n_ports;
-
 	ret = ahci_start_ports(uc_priv);
 	if (ret)
 		return ret;
@@ -1238,15 +1235,15 @@ int ahci_probe_scsi_pci(struct udevice *ahci_dev)
 	ulong base;
 	u16 vendor, device;
 
-	base = (uintptr_t)dm_pci_map_bar(ahci_dev, PCI_BASE_ADDRESS_5,
-					 PCI_REGION_MEM);
+	base = (ulong)dm_pci_map_bar(ahci_dev, PCI_BASE_ADDRESS_5,
+				     PCI_REGION_MEM);
 
 	dm_pci_read_config16(ahci_dev, PCI_VENDOR_ID, &vendor);
 	dm_pci_read_config16(ahci_dev, PCI_DEVICE_ID, &device);
 
 	if (vendor == 0x177d && device == 0xa01c)
-		base = (uintptr_t)dm_pci_map_bar(ahci_dev, PCI_BASE_ADDRESS_0,
-						 PCI_REGION_MEM);
+		base = (ulong)dm_pci_map_bar(ahci_dev, PCI_BASE_ADDRESS_0,
+					     PCI_REGION_MEM);
 	return ahci_probe_scsi(ahci_dev, base);
 }
 #endif
@@ -1273,3 +1270,4 @@ __weak int scsi_bus_reset(struct udevice *dev)
 
 	return 0;
 }
+#endif
diff --git a/drivers/block/Makefile b/drivers/block/Makefile
index b247735a7d..3feb0aa997 100644
--- a/drivers/block/Makefile
+++ b/drivers/block/Makefile
@@ -14,6 +14,3 @@ obj-$(CONFIG_IDE) += ide.o
 endif
 obj-$(CONFIG_SANDBOX) += sandbox.o
 obj-$(CONFIG_$(SPL_TPL_)BLOCK_CACHE) += blkcache.o
-obj-$(CONFIG_SATA_AHCI) += sata_ahci.o
-obj-$(CONFIG_NVME) += nvme.o
-obj-$(CONFIG_NVME) += nvme-uclass.o
diff --git a/drivers/block/blk-uclass.c b/drivers/block/blk-uclass.c
index 077a843700..baaf431e5e 100644
--- a/drivers/block/blk-uclass.c
+++ b/drivers/block/blk-uclass.c
@@ -3,6 +3,7 @@
  * Copyright (C) 2016 Google, Inc
  * Written by Simon Glass <sjg@chromium.org>
  */
+
 #include <common.h>
 #include <blk.h>
 #include <dm.h>
diff --git a/drivers/i2c/Kconfig b/drivers/i2c/Kconfig
index 8f0e9533a3..eb47454e76 100644
--- a/drivers/i2c/Kconfig
+++ b/drivers/i2c/Kconfig
@@ -367,11 +367,11 @@ config SYS_I2C_SANDBOX
 	  which specifies the driver to use.  See sandbox.dts as an example.
 
 config SYS_I2C_OCTEONTX
-	bool "OcteonTX i2c driver"
+	bool "OcteonTX I2C driver"
 	depends on (ARCH_OCTEONTX || ARCH_OCTEONTX2) && DM_I2C
 	default y
 	help
-	 Enable I2C support for OcteonTX line of processors.
+	 Enable I2C support for OcteonTX/TX2 line of processors.
 
 config SYS_I2C_S3C24X0
 	bool "Samsung I2C driver"
@@ -405,13 +405,6 @@ config SYS_I2C_TEGRA
 	help
 	  Support for NVIDIA I2C controller available in Tegra SoCs.
 
-config SYS_I2C_THUNDERX
-	bool "ThunderX i2c driver"
-	depends on ARCH_THUNDERX && DM_I2C
-	default y
-	help
-	 Enable I2C support for Cavium ThunderX line of processors.
-
 config SYS_I2C_UNIPHIER
 	bool "UniPhier I2C driver"
 	depends on ARCH_UNIPHIER && DM_I2C
diff --git a/drivers/i2c/octeontx_i2c.c b/drivers/i2c/octeontx_i2c.c
index 04c8199ed4..7d462ce28c 100644
--- a/drivers/i2c/octeontx_i2c.c
+++ b/drivers/i2c/octeontx_i2c.c
@@ -960,3 +960,9 @@ U_BOOT_DRIVER(octeontx_pci_twsi) = {
 	.ops	= &octeontx_i2c_ops,
 };
 
+static struct pci_device_id octeontx_twsi_supported[] = {
+	{ PCI_VDEVICE(CAVIUM, 0xa012) },
+	{ },
+};
+
+U_BOOT_PCI_DEVICE(octeontx_pci_twsi, octeontx_twsi_supported);
diff --git a/drivers/mmc/Kconfig b/drivers/mmc/Kconfig
index 655671cee3..19da6f88e2 100644
--- a/drivers/mmc/Kconfig
+++ b/drivers/mmc/Kconfig
@@ -13,10 +13,16 @@ config MMC
 
 config MMC_WRITE
 	bool "support for MMC/SD write operations"
+	depends on MMC
 	default y
 	help
 	  Enable write access to MMC and SD Cards
 
+config MMC_BROKEN_CD
+	bool "Poll for broken card detection case"
+	help
+	  If card  detection feature is broken, just poll to detect.
+
 config DM_MMC
 	bool "Enable MMC controllers using Driver Model"
 	depends on DM
@@ -289,7 +295,7 @@ config MMC_OCTEONTX
 	depends on DM_MMC
 	help
 	  This selects the OcteonTX Multimedia card Interface.
-	  If you have an OcteonTX board with a Multimedia Card slot,
+	  If you have an OcteonTX/TX2 board with a Multimedia Card slot,
 	  say Y here.
 
 	  If unsure, say N.
diff --git a/drivers/mmc/Makefile b/drivers/mmc/Makefile
index 8a295e3925..91398a9be0 100644
--- a/drivers/mmc/Makefile
+++ b/drivers/mmc/Makefile
@@ -3,9 +3,9 @@
 # (C) Copyright 2006
 # Wolfgang Denk, DENX Software Engineering, wd@denx.de.
 
+obj-y += mmc.o
 obj-$(CONFIG_$(SPL_)DM_MMC) += mmc-uclass.o
 obj-$(CONFIG_$(SPL_)MMC_WRITE) += mmc_write.o
-obj-$(CONFIG_MMC) += mmc.o
 
 ifndef CONFIG_$(SPL_)BLK
 obj-y += mmc_legacy.o
diff --git a/drivers/mmc/mmc-uclass.c b/drivers/mmc/mmc-uclass.c
index 81230f59a3..37c3843902 100644
--- a/drivers/mmc/mmc-uclass.c
+++ b/drivers/mmc/mmc-uclass.c
@@ -11,8 +11,6 @@
 #include <dm/lists.h>
 #include "mmc_private.h"
 
-DECLARE_GLOBAL_DATA_PTR;
-
 int dm_mmc_send_cmd(struct udevice *dev, struct mmc_cmd *cmd,
 		    struct mmc_data *data)
 {
@@ -196,9 +194,10 @@ int mmc_of_parse(struct udevice *dev, struct mmc_config *cfg)
 
 struct mmc *mmc_get_mmc_dev(struct udevice *dev)
 {
+	struct mmc_uclass_priv *upriv;
+
 	if (!device_active(dev))
 		return NULL;
-	struct mmc_uclass_priv *upriv;
 	upriv = dev_get_uclass_priv(dev);
 	return upriv->mmc;
 }
@@ -216,7 +215,7 @@ struct mmc *find_mmc_device(int dev_num)
 		printf("MMC Device %d not found\n", dev_num);
 #endif
 		return NULL;
-        }
+	}
 
 	mmc_dev = dev_get_parent(dev);
 
@@ -292,8 +291,7 @@ void print_mmc_devices(char separator)
 		else
 			mmc_type = NULL;
 
-		printf("%s: %d", m->cfg->name,
-			mmc_get_blk_desc(m)->devnum);
+		printf("%s: %d", m->cfg->name, mmc_get_blk_desc(m)->devnum);
 		if (mmc_type)
 			printf(" (%s)", mmc_type);
 	}
@@ -360,8 +358,10 @@ int mmc_unbind(struct udevice *dev)
 static int mmc_select_hwpart(struct udevice *bdev, int hwpart)
 {
 	struct udevice *mmc_dev = dev_get_parent(bdev);
-	struct blk_desc *desc = dev_get_uclass_platdata(bdev);
 	struct mmc *mmc = mmc_get_mmc_dev(mmc_dev);
+	struct blk_desc *desc = dev_get_uclass_platdata(bdev);
+	int ret;
+
 	if (desc->hwpart == hwpart)
 		return 0;
 
diff --git a/drivers/mmc/octeontx_hsmmc.c b/drivers/mmc/octeontx_hsmmc.c
index 04c6509735..79129db576 100644
--- a/drivers/mmc/octeontx_hsmmc.c
+++ b/drivers/mmc/octeontx_hsmmc.c
@@ -3149,45 +3149,6 @@ static int octeontx_mmc_host_probe(struct udevice *dev)
 	return 0;
 }
 
-static int octeontx_mmc_host_bind(struct udevice *parent)
-{
-	struct udevice *dev;
-	ofnode node;
-	int err;
-
-	debug("%s(%s): bind\n", __func__, parent->name);
-	dev_for_each_subnode(node, parent) {
-		struct uclass *uc;
-		int slot = ofnode_read_u32_default(node, "reg", -1);
-		char name[32];
-
-		if (slot < 0) {
-			pr_warn("Warning: mmc slot in device tree missing reg\n");
-			continue;
-		}
-		snprintf(name, sizeof(name), "octeontx-hsmmc%d", slot);
-		err = device_bind_driver_to_node(parent, "octeontx_hsmmc_slot",
-						 ofnode_get_name(node), node,
-						 &dev);
-		if (err) {
-			pr_err("%s(%s): Error %d binding slot %s\n", __func__,
-			       parent->name, err, name);
-			continue;
-		}
-		pr_info("%s: bound %s\n", parent->name, dev->name);
-		err = uclass_get(UCLASS_MMC, &uc);
-		if (err) {
-			pr_err("%s(%s: Could not get MMC uclass\n", __func__,
-			       parent->name);
-			continue;
-		}
-		dev->uclass = uc;
-		debug("  uclass priv: %p\n", uc->priv);
-	}
-
-	return 0;
-}
-
 /**
  * This performs some initial setup before a probe occurs.
  *
@@ -3257,7 +3218,6 @@ static const struct udevice_id octeontx_hsmmc_host_ids[] = {
 U_BOOT_DRIVER(octeontx_hsmmc_host) = {
 	.name	= "octeontx_hsmmc_host",
 	.id	= UCLASS_MISC,
-	.bind	= octeontx_mmc_host_bind,
 	.of_match = of_match_ptr(octeontx_hsmmc_host_ids),
 	.probe	= octeontx_mmc_host_probe,
 	.priv_auto_alloc_size = sizeof(struct octeontx_mmc_host),
@@ -3265,3 +3225,9 @@ U_BOOT_DRIVER(octeontx_hsmmc_host) = {
 	.flags	= DM_FLAG_PRE_RELOC,
 };
 
+static struct pci_device_id octeontx_mmc_supported[] = {
+	{ PCI_VDEVICE(CAVIUM, PCI_DEVICE_ID_OCTEONTX_EMMC) },
+	{ },
+};
+
+U_BOOT_PCI_DEVICE(octeontx_hsmmc_host, octeontx_mmc_supported);
diff --git a/drivers/mmc/octeontx_hsmmc.h b/drivers/mmc/octeontx_hsmmc.h
index 8cdc8f6bfc..0e2e143e5f 100644
--- a/drivers/mmc/octeontx_hsmmc.h
+++ b/drivers/mmc/octeontx_hsmmc.h
@@ -36,6 +36,8 @@
 
 #define MAX_NO_OF_TAPS			64
 
+#define EXT_CSD_POWER_CLASS		187	/* R/W */
+
 struct octeontx_mmc_host;
 
 /** MMC/SD slot data structure */
diff --git a/drivers/mtd/nand/Kconfig b/drivers/mtd/nand/Kconfig
index 5111200718..78ae04bdcb 100644
--- a/drivers/mtd/nand/Kconfig
+++ b/drivers/mtd/nand/Kconfig
@@ -4,4 +4,3 @@ config MTD_NAND_CORE
 source "drivers/mtd/nand/raw/Kconfig"
 
 source "drivers/mtd/nand/spi/Kconfig"
-
diff --git a/drivers/mtd/nand/raw/Kconfig b/drivers/mtd/nand/raw/Kconfig
index 1c03d4365c..10ac180199 100644
--- a/drivers/mtd/nand/raw/Kconfig
+++ b/drivers/mtd/nand/raw/Kconfig
@@ -123,21 +123,6 @@ config NAND_LPC32XX_SLC
 	help
 	  Enable the LPC32XX SLC NAND controller.
 
-config NAND_OCTEONTX
-	bool "Support for OcteonTX NAND controller"
-	select SYS_NAND_SELF_INIT
-	imply CMD_NAND
-	help
-	 This enables Nand flash controller found on the OcteonTX processors.
-
-config NAND_OCTEONTX_HW_ECC
-	bool "Support Hardware ECC for OcteonTX NAND controller"
-	depends on NAND_OCTEONTX
-	default y
-	help
-	 This enables Hardware BCH engine found on the OcteonTX processors to
-	 support ECC for NAND flash controller.
-
 config NAND_OMAP_GPMC
 	bool "Support OMAP GPMC NAND controller"
 	depends on ARCH_OMAP2PLUS
@@ -301,6 +286,22 @@ config NAND_ZYNQ_USE_BOOTLOADER1_TIMINGS
 	  This flag prevent U-boot reconfigure NAND flash controller and reuse
 	  the NAND timing from 1st stage bootloader.
 
+config NAND_OCTEONTX
+	bool "Support for OcteonTX NAND controller"
+	select SYS_NAND_SELF_INIT
+	imply CMD_NAND
+	help
+	 This enables Nand flash controller hardware found on the OcteonTX
+	 processors.
+
+config NAND_OCTEONTX_HW_ECC
+	bool "Support Hardware ECC for OcteonTX NAND controller"
+	depends on NAND_OCTEONTX
+	default y
+	help
+	 This enables Hardware BCH engine found on the OcteonTX processors to
+	 support ECC for NAND flash controller.
+
 config NAND_STM32_FMC2
 	bool "Support for NAND controller on STM32MP SoCs"
 	depends on ARCH_STM32MP
diff --git a/drivers/mtd/nand/octeontx_bch.c b/drivers/mtd/nand/raw/octeontx_bch.c
similarity index 100%
rename from drivers/mtd/nand/octeontx_bch.c
rename to drivers/mtd/nand/raw/octeontx_bch.c
diff --git a/drivers/mtd/nand/octeontx_bch.h b/drivers/mtd/nand/raw/octeontx_bch.h
similarity index 100%
rename from drivers/mtd/nand/octeontx_bch.h
rename to drivers/mtd/nand/raw/octeontx_bch.h
diff --git a/drivers/mtd/nand/octeontx_bch_regs.h b/drivers/mtd/nand/raw/octeontx_bch_regs.h
similarity index 100%
rename from drivers/mtd/nand/octeontx_bch_regs.h
rename to drivers/mtd/nand/raw/octeontx_bch_regs.h
diff --git a/drivers/mtd/nand/octeontx_nand.c b/drivers/mtd/nand/raw/octeontx_nand.c
similarity index 99%
rename from drivers/mtd/nand/octeontx_nand.c
rename to drivers/mtd/nand/raw/octeontx_nand.c
index 14f237331c..69b21e5ec0 100644
--- a/drivers/mtd/nand/octeontx_nand.c
+++ b/drivers/mtd/nand/raw/octeontx_nand.c
@@ -447,20 +447,6 @@ static inline struct octeontx_nfc *to_otx_nfc(struct nand_hw_control *ctrl)
 	return container_of(ctrl, struct octeontx_nfc, controller);
 }
 
-/** Ported from Linux 4.9.0 include/linux/mtd/mtd.h for compatibility */
-static inline void mtd_set_of_node(struct mtd_info *mtd, ofnode node)
-{
-	mtd->dev->node = node;
-}
-
-/** Ported from Linux 4.9.0 include/linux/mtd/nand.h for compatibility */
-static inline void nand_set_flash_node(struct nand_chip *chip,
-				       ofnode node)
-{
-	mtd_set_of_node(&chip->mtd, node);
-	chip->flash_node = ofnode_to_offset(node);
-}
-
 #if defined(CONFIG_NAND_OCTEONTX_HW_ECC)
 static int octeontx_nand_calc_ecc_layout(struct nand_chip *nand)
 {
diff --git a/drivers/net/e1000.c b/drivers/net/e1000.c
index ef4d71ff0d..2a5d09d143 100644
--- a/drivers/net/e1000.c
+++ b/drivers/net/e1000.c
@@ -5789,7 +5789,6 @@ static int do_e1000(cmd_tbl_t *cmdtp, int flag,
 		plat = dev_get_platdata(dev);
 		mac = plat->enetaddr;
 	}
-	hw = dev_get_priv(dev);
 #else
 	hw = e1000_find_card(cardnum);
 	if (hw)
diff --git a/drivers/net/e1000_spi.c b/drivers/net/e1000_spi.c
index f6ae3c4d07..aecd290d72 100644
--- a/drivers/net/e1000_spi.c
+++ b/drivers/net/e1000_spi.c
@@ -104,14 +104,14 @@ struct spi_slave *spi_setup_slave(unsigned int bus, unsigned int cs,
 	}
 
 	/* TODO: Use max_hz somehow */
-	E1000_DBG(hw, "EEPROM SPI access requested\n");
+	E1000_DBG(hw->nic, "EEPROM SPI access requested\n");
 	return &hw->spi;
 }
 
 void spi_free_slave(struct spi_slave *spi)
 {
 	__maybe_unused struct e1000_hw *hw = e1000_hw_from_spi(spi);
-	E1000_DBG(hw, "EEPROM SPI access released\n");
+	E1000_DBG(hw->nic, "EEPROM SPI access released\n");
 }
 
 int spi_claim_bus(struct spi_slave *spi)
diff --git a/drivers/net/octeontx/nic_main.c b/drivers/net/octeontx/nic_main.c
index e821dbd3f4..b541926e0d 100644
--- a/drivers/net/octeontx/nic_main.c
+++ b/drivers/net/octeontx/nic_main.c
@@ -20,8 +20,6 @@
 #include "nic.h"
 #include "q_struct.h"
 
-#define ETH_HLEN 6
-
 unsigned long rounddown_pow_of_two(unsigned long n)
 {
 	n |= n >> 1;
diff --git a/drivers/net/octeontx/nicvf_main.c b/drivers/net/octeontx/nicvf_main.c
index 4a10e69870..775bbc515f 100644
--- a/drivers/net/octeontx/nicvf_main.c
+++ b/drivers/net/octeontx/nicvf_main.c
@@ -18,8 +18,6 @@
 #include "nic.h"
 #include "nicvf_queues.h"
 
-#define ETH_ALEN 6
-
 /* Register read/write APIs */
 void nicvf_reg_write(struct nicvf *nic, u64 offset, u64 val)
 {
diff --git a/drivers/net/octeontx/smi.c b/drivers/net/octeontx/smi.c
index d7ef8547e6..4abe41cd01 100644
--- a/drivers/net/octeontx/smi.c
+++ b/drivers/net/octeontx/smi.c
@@ -13,7 +13,7 @@
 #include <misc.h>
 #include <malloc.h>
 #include <asm/io.h>
-#include <environment.h>
+#include <env.h>
 #include <linux/ctype.h>
 
 #define PCI_DEVICE_ID_OCTEONTX_SMI 0xA02B
diff --git a/drivers/net/octeontx2/nix_af.c b/drivers/net/octeontx2/nix_af.c
index 7bcae3bd15..fe437ce0f7 100644
--- a/drivers/net/octeontx2/nix_af.c
+++ b/drivers/net/octeontx2/nix_af.c
@@ -878,7 +878,7 @@ int npc_lf_admin_setup(struct nix *nix)
 			 NPC_AF_INTFX_KEX_CFG(NPC_INTF_E_NIXX_RX(0)),
 			 kex_cfg.u);
 
-	/* Errata #35786 */
+	/* HW Issue */
 	kex_cfg.u = 0;
 	kex_cfg.s.parse_nibble_ena = 0x7;
 	npc_af_reg_write(nix_af,
@@ -1021,7 +1021,7 @@ int nix_af_setup(struct nix_af *nix_af)
 		return -1;
 	}
 	af_cfg.s.af_be = 0;
-	af_cfg.u |= 0x5E;	/* Errata 35057 */
+	af_cfg.u |= 0x5E;	/* HW Issue */
 	nix_af_reg_write(nix_af, NIXX_AF_CFG(), af_cfg.u);
 
 	/* Perform Calibration */
diff --git a/drivers/pci/pci_auto.c b/drivers/pci/pci_auto.c
index 4ad102a69c..1a3bf70834 100644
--- a/drivers/pci/pci_auto.c
+++ b/drivers/pci/pci_auto.c
@@ -11,7 +11,6 @@
 #include <dm.h>
 #include <errno.h>
 #include <pci.h>
-#include <dm/device-internal.h>
 
 /* the user can define CONFIG_SYS_PCI_CACHE_LINE_SIZE to avoid problems */
 #ifndef CONFIG_SYS_PCI_CACHE_LINE_SIZE
@@ -317,7 +316,6 @@ int dm_pciauto_config_device(struct udevice *dev)
 	struct pci_controller *ctlr_hose = dev_get_uclass_priv(ctlr);
 	int n;
 
-//	enum_only = dm_pci_find_capability(dev, PCI_CAP_ID_EA);
 #ifdef CONFIG_PCI_ENUM_ONLY
 	enum_only = true;
 #endif
diff --git a/drivers/spi/spi-uclass.c b/drivers/spi/spi-uclass.c
index 8bc3800cc2..76c4b53c16 100644
--- a/drivers/spi/spi-uclass.c
+++ b/drivers/spi/spi-uclass.c
@@ -92,30 +92,6 @@ int dm_spi_xfer(struct udevice *dev, unsigned int bitlen,
 	return spi_get_ops(bus)->xfer(dev, bitlen, dout, din, flags);
 }
 
-bool dm_spi_is_flash_command_supported(struct udevice *dev,
-				       const struct spi_flash_command *cmd)
-{
-	struct udevice *bus = dev->parent;
-	struct dm_spi_ops *ops = spi_get_ops(bus);
-
-	if (ops->is_flash_command_supported)
-		return ops->is_flash_command_supported(dev, cmd);
-
-	return false;
-}
-
-int dm_spi_exec_flash_command(struct udevice *dev,
-			      const struct spi_flash_command *cmd)
-{
-	struct udevice *bus = dev->parent;
-	struct dm_spi_ops *ops = spi_get_ops(bus);
-
-	if (ops->exec_flash_command)
-		return ops->exec_flash_command(dev, cmd);
-
-	return -EINVAL;
-}
-
 int spi_claim_bus(struct spi_slave *slave)
 {
 	return log_ret(dm_spi_claim_bus(slave->dev));
@@ -192,10 +168,6 @@ static int spi_post_probe(struct udevice *bus)
 		ops->set_mode += gd->reloc_off;
 	if (ops->cs_info)
 		ops->cs_info += gd->reloc_off;
-	if (ops->is_flash_command_supported)
-		ops->is_flash_command_supported += gd->reloc_off;
-	if (ops->exec_flash_command)
-		ops->exec_flash_command += gd->reloc_off;
 #endif
 
 	return 0;
@@ -324,7 +296,6 @@ int spi_get_bus_and_cs(int busnum, int cs, int speed, int mode,
 {
 	struct udevice *bus, *dev;
 	struct dm_spi_slave_platdata *plat;
-	struct dm_spi_bus *spi;
 	bool created = false;
 	int ret;
 
@@ -338,7 +309,6 @@ int spi_get_bus_and_cs(int busnum, int cs, int speed, int mode,
 		return ret;
 	}
 	ret = spi_find_chip_select(bus, cs, &dev);
-	spi = dev_get_uclass_priv(bus);
 
 	/*
 	 * If there is no such device, create one automatically. This means
@@ -388,9 +358,6 @@ int spi_get_bus_and_cs(int busnum, int cs, int speed, int mode,
 		speed = plat->max_hz;
 		mode = plat->mode;
 	}
-	if (spi->max_hz)
-		speed = min(speed, (int)spi->max_hz);
-
 	ret = spi_set_speed_mode(bus, speed, mode);
 	if (ret)
 		goto err;
diff --git a/drivers/spi/spi.c b/drivers/spi/spi.c
index aac2a17f9e..22910de0dd 100644
--- a/drivers/spi/spi.c
+++ b/drivers/spi/spi.c
@@ -4,7 +4,6 @@
  */
 
 #include <common.h>
-#include <errno.h>
 #include <fdtdec.h>
 #include <malloc.h>
 #include <spi.h>
diff --git a/env/mmc.c b/env/mmc.c
index 4b97e11cce..9f1878def1 100644
--- a/env/mmc.c
+++ b/env/mmc.c
@@ -142,12 +142,7 @@ static int mmc_set_env_part(struct mmc *mmc)
 	int dev = mmc_get_env_dev();
 	int ret = 0;
 
-	env_mmc_orig_hwpart =
-#ifdef CONFIG_MMC_OCTEONTX
-		mmc_get_blk_desc(mmc, CONFIG_SYS_MMC_ENV_DEV)->hwpart;
-#else
-		mmc_get_blk_desc(mmc)->hwpart;
-#endif
+	env_mmc_orig_hwpart = mmc_get_blk_desc(mmc)->hwpart;
 	ret = blk_select_hwpart_devnum(IF_TYPE_MMC, dev, part);
 	if (ret)
 		puts("MMC partition switch failed\n");
@@ -192,11 +187,8 @@ static inline int write_env(struct mmc *mmc, unsigned long size,
 			    unsigned long offset, const void *buffer)
 {
 	uint blk_start, blk_cnt, n;
-#ifdef CONFIG_MMC_OCTEONTX
-	struct blk_desc *desc = mmc_get_blk_desc(mmc, CONFIG_SYS_MMC_ENV_DEV);
-#else
 	struct blk_desc *desc = mmc_get_blk_desc(mmc);
-#endif
+
 	blk_start	= ALIGN(offset, mmc->write_bl_len) / mmc->write_bl_len;
 	blk_cnt		= ALIGN(size, mmc->write_bl_len) / mmc->write_bl_len;
 
@@ -305,11 +297,8 @@ static inline int read_env(struct mmc *mmc, unsigned long size,
 			   unsigned long offset, const void *buffer)
 {
 	uint blk_start, blk_cnt, n;
-#ifdef CONFIG_MMC_OCTEONTX
-	struct blk_desc *desc = mmc_get_blk_desc(mmc, CONFIG_SYS_MMC_ENV_DEV);
-#else
 	struct blk_desc *desc = mmc_get_blk_desc(mmc);
-#endif
+
 	blk_start	= ALIGN(offset, mmc->read_bl_len) / mmc->read_bl_len;
 	blk_cnt		= ALIGN(size, mmc->read_bl_len) / mmc->read_bl_len;
 
diff --git a/env/sf.c b/env/sf.c
index f679236ffa..8313193bd5 100644
--- a/env/sf.c
+++ b/env/sf.c
@@ -68,7 +68,7 @@ extern void board_get_env_spi_bus_cs(int *bus, int *cs);
 				     CONFIG_ENV_SPI_MAX_HZ, CONFIG_ENV_SPI_MODE,
 				     &new);
 	if (ret) {
-		set_default_env("!spi_flash_probe_bus_cs() failed");
+		env_set_default("spi_flash_probe_bus_cs() failed", 0);
 		return ret;
 	}
 #endif
diff --git a/include/common.h b/include/common.h
index 75c70a4a12..d8f302ea92 100644
--- a/include/common.h
+++ b/include/common.h
@@ -182,7 +182,6 @@ int board_fix_fdt (void *rw_fdt_blob); /* manipulate the U-Boot fdt before its r
 int board_late_init (void);
 int board_postclk_init (void); /* after clocks/timebase, before env/serial */
 int board_early_init_r (void);
-bool alternate_pkg (void);
 
 #if defined(CONFIG_SYS_DRAM_TEST)
 int testdram(void);
diff --git a/include/ddr_spd.h b/include/ddr_spd.h
index e4313ce4b5..fe163da43e 100644
--- a/include/ddr_spd.h
+++ b/include/ddr_spd.h
@@ -504,8 +504,6 @@ extern void ddr2_spd_dump(const ddr2_spd_eeprom_t *spd);
 extern unsigned int ddr3_spd_check(const ddr3_spd_eeprom_t *spd);
 unsigned int ddr4_spd_check(const struct ddr4_spd_eeprom_s *spd);
 
-const char *ddr3_modtype_str(const ddr3_spd_eeprom_t *spd);
-
 /*
  * Byte 2 Fundamental Memory Types.
  */
diff --git a/include/mmc.h b/include/mmc.h
index a612f73b94..a76667aa60 100644
--- a/include/mmc.h
+++ b/include/mmc.h
@@ -39,7 +39,6 @@
 #define EXTRACT_SDMMC_CHANGE_VERSION(x)	\
 	((u32)(x) & 0xff)
 
-#define SD_VERSION_4		MAKE_SD_VERSION(4, 0, 0)
 #define SD_VERSION_3		MAKE_SD_VERSION(3, 0, 0)
 #define SD_VERSION_2		MAKE_SD_VERSION(2, 0, 0)
 #define SD_VERSION_1_0		MAKE_SD_VERSION(1, 0, 0)
@@ -158,13 +157,6 @@ static inline bool mmc_is_tuning_cmd(uint cmdidx)
 #define OCR_BUSY		0x80000000
 #define OCR_HCS			0x40000000
 #define OCR_S18R		0x1000000
-#define MMC_HS_TIMING		0x00000100
-#define MMC_HS_DDR_52MHz_12V	0x8
-#define MMC_HS_DDR_52MHz_18_3V	0x4
-#define MMC_HS_52MHZ		0x2
-#define MMC_HS_26MHz		0x1
-
-#define OCR_XPC			0x10000000
 #define OCR_VOLTAGE_MASK	0x007FFF80
 #define OCR_ACCESS_MODE		0x60000000
 
@@ -234,25 +226,15 @@ static inline bool mmc_is_tuning_cmd(uint cmdidx)
 #define EXT_CSD_BUS_WIDTH		183	/* R/W */
 #define EXT_CSD_STROBE_SUPPORT		184	/* R/W */
 #define EXT_CSD_HS_TIMING		185	/* R/W */
-#define EXT_CSD_POWER_CLASS		187	/* R/W */
 #define EXT_CSD_REV			192	/* RO */
 #define EXT_CSD_CARD_TYPE		196	/* RO */
 #define EXT_CSD_PART_SWITCH_TIME	199	/* RO */
-#define EXT_CSD_PWR_CL_52_195		200	/* RO */
-#define EXT_CSD_PWR_CL_26_195		201	/* RO */
-#define EXT_CSD_PWR_CL_52_360		202	/* RO */
-#define EXT_CSD_PWR_CL_26_360		203	/* RO */
 #define EXT_CSD_SEC_CNT			212	/* RO, 4 bytes */
 #define EXT_CSD_HC_WP_GRP_SIZE		221	/* RO */
 #define EXT_CSD_HC_ERASE_GRP_SIZE	224	/* RO */
 #define EXT_CSD_BOOT_MULT		226	/* RO */
 #define EXT_CSD_GENERIC_CMD6_TIME       248     /* RO */
 #define EXT_CSD_BKOPS_SUPPORT		502	/* RO */
-#define EXT_CSD_PWR_CL_200_195		236	/* RO */
-#define EXT_CSD_PWR_CL_200_360		237	/* RO */
-#define EXT_CSD_PWR_CL_DDR_52_195	238	/* RO */
-#define EXT_CSD_PWR_CL_DDR_52_360	239	/* RO */
-#define EXT_CSD_GENERIC_CMD6_TIME	248	/* RO in 10ms increments */
 
 /*
  * EXT_CSD field definitions
@@ -266,8 +248,6 @@ static inline bool mmc_is_tuning_cmd(uint cmdidx)
 #define EXT_CSD_CARD_TYPE_52	(1 << 1)	/* Card can run at 52MHz */
 #define EXT_CSD_CARD_TYPE_DDR_1_8V	(1 << 2)
 #define EXT_CSD_CARD_TYPE_DDR_1_2V	(1 << 3)
-#define EXT_CSD_CARD_TYPE_HS400_1_8V	(1 << 6)	/* HS400 DDR, 1.8V */
-#define EXT_CSD_CARD_TYPE_HS400_1_2V	(1 << 7)	/* HS400 DDR, 1.2V */
 #define EXT_CSD_CARD_TYPE_DDR_52	(EXT_CSD_CARD_TYPE_DDR_1_8V \
 					| EXT_CSD_CARD_TYPE_DDR_1_2V)
 
@@ -753,15 +733,11 @@ int mmc_voltage_to_mv(enum mmc_voltage voltage);
  * @disable:	flag indicating if the clock must on or off
  * @return 0 if OK, -ve on error
  */
-#if !CONFIG_IS_ENABLED(MMC_OCTEONTX)
 int mmc_set_clock(struct mmc *mmc, uint clock, bool disable);
 
 #define MMC_CLK_ENABLE		false
 #define MMC_CLK_DISABLE		true
 
-#else
-void mmc_set_clock(struct mmc *mmc, uint clock);
-#endif
 struct mmc *find_mmc_device(int dev_num);
 int mmc_set_dev(int dev_num);
 void print_mmc_devices(char separator);
@@ -886,12 +862,8 @@ int mmc_power_init(struct mmc *mmc);
  * mmc_get_blk_desc() - Get the block descriptor for an MMC device
  *
  * @mmc:	MMC device
- * @devnum:	device number
  * @return block device if found, else NULL
  */
-#ifdef CONFIG_MMC_OCTEONTX
-struct blk_desc *mmc_get_blk_desc(struct mmc *mmc, int devnum);
-#else
 struct blk_desc *mmc_get_blk_desc(struct mmc *mmc);
-#endif
+
 #endif /* _MMC_H_ */
diff --git a/include/pci.h b/include/pci.h
index cf5fe93515..85163baea9 100644
--- a/include/pci.h
+++ b/include/pci.h
@@ -461,6 +461,9 @@
 #define PCI_EA_FIRST_ENT	4	/* First EA Entry in List */
 #define  PCI_EA_ES		0x00000007 /* Entry Size */
 #define  PCI_EA_BEI		0x000000f0 /* BAR Equivalent Indicator */
+/* 9-14 map to VF BARs 0-5 respectively */
+#define  PCI_EA_BEI_VF_BAR0	9
+#define  PCI_EA_BEI_VF_BAR5	14
 /* Base, MaxOffset registers */
 /* bit 0 is reserved */
 #define  PCI_EA_IS_64		0x00000002	/* 64-bit field flag */
@@ -471,37 +474,17 @@
 #define  PCI_EXP_DEVCAP_FLR     0x10000000 /* Function Level Reset */
 #define PCI_EXP_DEVCTL		8	/* Device Control */
 #define  PCI_EXP_DEVCTL_BCR_FLR 0x8000  /* Bridge Configuration Retry / FLR */
-
-/* Enhanced Allocation Registers */
-#define PCI_EA_NUM_ENT		2	/* Number of Capability Entries */
-#define  PCI_EA_NUM_ENT_MASK	0x3f	/* Num Entries Mask */
-#define PCI_EA_FIRST_ENT	4	/* First EA Entry in List */
-#define  PCI_EA_ES		0x00000007 /* Entry Size */
-#define  PCI_EA_BEI		0x000000f0 /* BAR Equivalent Indicator */
-/* 9-14 map to VF BARs 0-5 respectively */
-#define  PCI_EA_BEI_VF_BAR0	9
-#define  PCI_EA_BEI_VF_BAR5	14
-/* Base, MaxOffset registers */
-/* bit 0 is reserved */
-#define  PCI_EA_IS_64		0x00000002	/* 64-bit field flag */
-#define  PCI_EA_FIELD_MASK	0xfffffffc	/* For Base & Max Offset */
-
-/* Single Root I/O Virtualization */
+/* Single Root I/O Virtualization Registers */
 #define PCI_SRIOV_CAP		0x04	/* SR-IOV Capabilities */
 #define PCI_SRIOV_CTRL		0x08	/* SR-IOV Control */
 #define  PCI_SRIOV_CTRL_VFE	0x01	/* VF Enable */
-#define  PCI_SRIOV_CTRL_VFM	0x02	/* VF Migration Enable */
 #define  PCI_SRIOV_CTRL_MSE	0x08	/* VF Memory Space Enable */
-#define PCI_SRIOV_STATUS	0x0a	/* SR-IOV Status */
-#define  PCI_SRIOV_STATUS_VFM	0x01	/* VF Migration Status */
 #define PCI_SRIOV_INITIAL_VF	0x0c	/* Initial VFs */
 #define PCI_SRIOV_TOTAL_VF	0x0e	/* Total VFs */
 #define PCI_SRIOV_NUM_VF	0x10	/* Number of VFs */
 #define PCI_SRIOV_VF_OFFSET	0x14	/* First VF Offset */
 #define PCI_SRIOV_VF_STRIDE	0x16	/* Following VF Stride */
 #define PCI_SRIOV_VF_DID	0x1a	/* VF Device ID */
-#define PCI_SRIOV_BAR		0x24	/* VF BAR0 */
-#define  PCI_SRIOV_NUM_BARS	6	/* Number of VF BARs */
 
 /* Include the ID list */
 
@@ -802,6 +785,8 @@ extern int pci_hose_find_cap_start(struct pci_controller *hose, pci_dev_t dev,
 extern int pci_find_cap(struct pci_controller *hose, pci_dev_t dev, int pos,
 			int cap);
 
+int pci_find_next_ext_capability(struct pci_controller *hose,
+				 pci_dev_t dev, int start, int cap);
 int pci_hose_find_ext_capability(struct pci_controller *hose,
 				 pci_dev_t dev, int cap);
 
@@ -1372,7 +1357,7 @@ pci_addr_t dm_pci_phys_to_bus(struct udevice *dev, phys_addr_t addr,
  * 32b/64b BARs.  Note that duplicate BEI entries are not supported.
  *
  * @dev:	Device to check
- * @bar:	Bar number to read (numbered from 0)
+ * @bar:	Bar register offset (PCI_BASE_ADDRESS_...)
  * @flags:	Flags for the region type (PCI_REGION_...)
  * @return: pointer to the virtual address to use or 0 on error
  */
@@ -1503,25 +1488,6 @@ int dm_pci_flr(struct udevice *dev);
 #define dm_pci_io_to_virt(dev, addr, len, map_flags) \
 	dm_pci_bus_to_virt((dev), (addr), PCI_REGION_IO, (len), (map_flags))
 
-/**
- * dm_pci_find_capability() - find a capability offset for a given device
- *
- * @dev: 	Device to check
- * @cap:	Capability ID
- * @return: capability offset if found, 0 otherwise
- */
-int dm_pci_find_capability(struct udevice *dev, int cap);
-
-/**
- * dm_pci_find_ext_capability() - find an extended capability
- * 		offset for a given device
- *
- * @dev: 	Device to check
- * @cap:	Capability ID
- * @return: capability offset if found, 0 otherwise
- */
-int dm_pci_find_ext_capability(struct udevice *dev, int cap);
-
 /**
  * dm_pci_find_device() - find a device by vendor/device ID
  *
diff --git a/include/pci_ids.h b/include/pci_ids.h
index e6916f38f6..f8b4f28996 100644
--- a/include/pci_ids.h
+++ b/include/pci_ids.h
@@ -2360,6 +2360,8 @@
 #define PCI_DEVICE_ID_ALTIMA_AC9100	0x03ea
 #define PCI_DEVICE_ID_ALTIMA_AC1003	0x03eb
 
+#define PCI_VENDOR_ID_CAVIUM		0x177d
+
 #define PCI_VENDOR_ID_BELKIN		0x1799
 #define PCI_DEVICE_ID_BELKIN_F5D7010V7	0x701f
 
@@ -3134,6 +3136,3 @@
 #define PCI_DEVICE_ID_XEN_PLATFORM	0x0001
 
 #define PCI_VENDOR_ID_OCZ		0x1b85
-
-#define PCI_VENDOR_ID_CAVIUM		0x177d
-
diff --git a/include/phy.h b/include/phy.h
index fa3b29232f..b71c296b08 100644
--- a/include/phy.h
+++ b/include/phy.h
@@ -54,60 +54,6 @@
 #endif
 
 
-typedef enum {
-	PHY_INTERFACE_MODE_MII,
-	PHY_INTERFACE_MODE_GMII,
-	PHY_INTERFACE_MODE_SGMII,
-	PHY_INTERFACE_MODE_SGMII_2500,
-	PHY_INTERFACE_MODE_QSGMII,
-	PHY_INTERFACE_MODE_TBI,
-	PHY_INTERFACE_MODE_RMII,
-	PHY_INTERFACE_MODE_RGMII,
-	PHY_INTERFACE_MODE_RGMII_ID,
-	PHY_INTERFACE_MODE_RGMII_RXID,
-	PHY_INTERFACE_MODE_RGMII_TXID,
-	PHY_INTERFACE_MODE_RTBI,
-	PHY_INTERFACE_MODE_XGMII,
-	PHY_INTERFACE_MODE_XAUI,
-	PHY_INTERFACE_MODE_RXAUI,
-	PHY_INTERFACE_MODE_SFI,
-	PHY_INTERFACE_MODE_XLAUI,
-	PHY_INTERFACE_MODE_NONE,	/* Must be last */
-
-	PHY_INTERFACE_MODE_COUNT,
-} phy_interface_t;
-
-static const char *phy_interface_strings[] = {
-	[PHY_INTERFACE_MODE_MII]		= "mii",
-	[PHY_INTERFACE_MODE_GMII]		= "gmii",
-	[PHY_INTERFACE_MODE_SGMII]		= "sgmii",
-	[PHY_INTERFACE_MODE_SGMII_2500]		= "sgmii-2500",
-	[PHY_INTERFACE_MODE_QSGMII]		= "qsgmii",
-	[PHY_INTERFACE_MODE_TBI]		= "tbi",
-	[PHY_INTERFACE_MODE_RMII]		= "rmii",
-	[PHY_INTERFACE_MODE_RGMII]		= "rgmii",
-	[PHY_INTERFACE_MODE_RGMII_ID]		= "rgmii-id",
-	[PHY_INTERFACE_MODE_RGMII_RXID]		= "rgmii-rxid",
-	[PHY_INTERFACE_MODE_RGMII_TXID]		= "rgmii-txid",
-	[PHY_INTERFACE_MODE_RTBI]		= "rtbi",
-	[PHY_INTERFACE_MODE_XGMII]		= "xgmii",
-	[PHY_INTERFACE_MODE_XAUI]		= "xaui",
-	[PHY_INTERFACE_MODE_RXAUI]		= "rxaui",
-	[PHY_INTERFACE_MODE_SFI]		= "sfi",
-	[PHY_INTERFACE_MODE_XLAUI]		= "xlaui",
-	[PHY_INTERFACE_MODE_NONE]		= "",
-};
-
-static inline const char *phy_string_for_interface(phy_interface_t i)
-{
-	/* Default to unknown */
-	if (i > PHY_INTERFACE_MODE_NONE)
-		i = PHY_INTERFACE_MODE_NONE;
-
-	return phy_interface_strings[i];
-}
-
-
 struct phy_device;
 
 #define MDIO_NAME_LEN 32
diff --git a/include/spi.h b/include/spi.h
index 244ff014ca..5eec0c4775 100644
--- a/include/spi.h
+++ b/include/spi.h
@@ -64,116 +64,6 @@ struct dm_spi_slave_platdata {
 
 #endif /* CONFIG_DM_SPI */
 
-/**
- * enum spi_flash_protocol - SPI flash command protocol
- */
-#define SPI_FPROTO_INST_SHIFT	16
-#define SPI_FPROTO_INST_MASK	GENMASK(23, 16)
-#define SPI_FPROTO_INST(nbits)					\
-	((((unsigned long)(nbits)) << SPI_FPROTO_INST_SHIFT) &	\
-	 SPI_FPROTO_INST_MASK)
-
-#define SPI_FPROTO_ADDR_SHIFT	8
-#define SPI_FPROTO_ADDR_MASK	GENMASK(15, 8)
-#define SPI_FPROTO_ADDR(nbits)					\
-	((((unsigned long)(nbits)) << SPI_FPROTO_ADDR_SHIFT) &	\
-	 SPI_FPROTO_ADDR_MASK)
-
-#define SPI_FPROTO_DATA_SHIFT	0
-#define SPI_FPROTO_DATA_MASK	GENMASK(7, 0)
-#define SPI_FPROTO_DATA(nbits)					\
-	((((unsigned long)(nbits)) << SPI_FPROTO_DATA_SHIFT) &	\
-	 SPI_FPROTO_DATA_MASK)
-
-#define SPI_FPROTO(inst_nbits, addr_nbits, data_nbits)	\
-	(SPI_FPROTO_INST(inst_nbits) |			\
-	 SPI_FPROTO_ADDR(addr_nbits) |			\
-	 SPI_FPROTO_DATA(data_nbits))
-
-enum spi_flash_protocol {
-	SPI_FPROTO_1_1_1 = SPI_FPROTO(1, 1, 1),
-	SPI_FPROTO_1_1_2 = SPI_FPROTO(1, 1, 2),
-	SPI_FPROTO_1_1_4 = SPI_FPROTO(1, 1, 4),
-	SPI_FPROTO_1_2_2 = SPI_FPROTO(1, 2, 2),
-	SPI_FPROTO_1_4_4 = SPI_FPROTO(1, 4, 4),
-	SPI_FPROTO_2_2_2 = SPI_FPROTO(2, 2, 2),
-	SPI_FPROTO_4_4_4 = SPI_FPROTO(4, 4, 4),
-};
-
-static inline
-u8 spi_flash_protocol_get_inst_nbits(enum spi_flash_protocol proto)
-{
-	return ((unsigned long)(proto & SPI_FPROTO_INST_MASK)) >>
-		SPI_FPROTO_INST_SHIFT;
-}
-
-static inline
-u8 spi_flash_protocol_get_addr_nbits(enum spi_flash_protocol proto)
-{
-	return ((unsigned long)(proto & SPI_FPROTO_ADDR_MASK)) >>
-		SPI_FPROTO_ADDR_SHIFT;
-}
-
-static inline
-u8 spi_flash_protocol_get_data_nbits(enum spi_flash_protocol proto)
-{
-	return ((unsigned long)(proto & SPI_FPROTO_DATA_MASK)) >>
-		SPI_FPROTO_DATA_SHIFT;
-}
-
-/**
- * struct spi_flash_command - SPI flash command structure
- *
- * @instr:		Opcode sent to the SPI slave during instr clock cycles.
- * @mode:		Value sent to the SPI slave during mode clock cycles.
- * @num_mode_cycles:	Number of mode clock cycles.
- * @num_wait_states:	Number of wait-state clock cycles.
- * @addr_len:		Number of bytes sent during address clock cycles:
- *			should be 0, 3, or 4.
- * @addr:		Value sent to the SPI slave during address clock cycles.
- * @data_len:		Number of bytes to be sent during data clock cycles.
- * @tx_data:		Data sent to the SPI slave during data clock cycles.
- * @rx_data:		Data read from the SPI slave during data clock cycles.
- */
-struct spi_flash_command {
-	enum spi_flash_protocol proto;
-	u8 flags;
-#define SPI_FCMD_TYPE		GENMASK(2, 0)
-#define SPI_FCMD_READ		(0x0U << 0)
-#define SPI_FCMD_WRITE		(0x1U << 0)
-#define SPI_FCMD_ERASE		(0x2U << 0)
-#define SPI_FCMD_READ_REG	(0x3U << 0)
-#define SPI_FCMD_WRITE_REG	(0x4U << 0)
-
-	u8 inst;
-	u8 mode;
-	u8 num_mode_cycles;
-	u8 num_wait_states;
-	u8 addr_len;
-	u32 addr;
-	size_t data_len;
-	const void *tx_data;
-	void *rx_data;
-};
-
-/**
- * Initialize a 'struct spi_flash_command'.
- *
- * @cmd:		Pointer to the 'struct spi_flash_command' to initialize.
- * @instr:		Instruction opcode.
- * @addr_len:		Number of address bytes.
- */
-static inline void
-spi_flash_command_init(struct spi_flash_command *cmd,
-		       u8 inst, u8 addr_len, u8 flags)
-{
-	memset(cmd, 0, sizeof(*cmd));
-	cmd->proto = SPI_FPROTO_1_1_1;
-	cmd->inst = inst;
-	cmd->addr_len = addr_len;
-	cmd->flags = flags;
-}
-
 /**
  * struct spi_slave - Representation of a SPI slave
  *
@@ -572,26 +462,6 @@ struct dm_spi_ops {
 	 *	   is invalid, other -ve value on error
 	 */
 	int (*cs_info)(struct udevice *bus, uint cs, struct spi_cs_info *info);
-
-	/**
-	 * Check whether the given SPI flash command is supported.
-	 *
-	 * @bus:	The SPI bus
-	 * @cmd:	The SPI flash command to check.
-	 * @return:	true if supported, false otherwise
-	 */
-	bool (*is_flash_command_supported)(struct udevice *bus,
-					   const struct spi_flash_command *cmd);
-
-	/**
-	 * Execute a SPI flash command
-	 *
-	 * @bus:	The SPI bus
-	 * @cmd:	The SPI flash command to execute.
-	 * @return 0 if OK, -ve on error
-	 */
-	int (*exec_flash_command)(struct udevice *bus,
-				  const struct spi_flash_command *cmd);
 };
 
 struct dm_spi_emul_ops {
@@ -780,24 +650,6 @@ void dm_spi_release_bus(struct udevice *dev);
 int dm_spi_xfer(struct udevice *dev, unsigned int bitlen,
 		const void *dout, void *din, unsigned long flags);
 
-/**
- * Check whether the given SPI flash command is supported.
- *
- * @dev:	The SPI slave device
- * @cmd:	The SPI flash command
- */
-bool dm_spi_is_flash_command_supported(struct udevice *dev,
-				       const struct spi_flash_command *cmd);
-
-/**
- * Execute the given SPI flash command.
- *
- * @dev:	The SPI slave device
- * @cmd:	The SPI flash command
- */
-int dm_spi_exec_flash_command(struct udevice *dev,
-			      const struct spi_flash_command *cmd);
-
 /* Access the operations for a SPI device */
 #define spi_get_ops(dev)	((struct dm_spi_ops *)(dev)->driver->ops)
 #define spi_emul_get_ops(dev)	((struct dm_spi_emul_ops *)(dev)->driver->ops)
diff --git a/net/Kconfig b/net/Kconfig
index 9f712845c3..68cecf75a2 100644
--- a/net/Kconfig
+++ b/net/Kconfig
@@ -35,32 +35,4 @@ config TFTP_BLOCKSIZE
 	help
 	  Default TFTP block size.
 
-config BOOTP_PXE
-	bool "Enable BOOTP PXE"
-	depends on CMD_NET
-
-config BOOTP_SUBNETMASK
-	bool "Enable BOOTP subnetmask"
-	depends on CMD_NET
-	depends on CMD_NET
-
-config BOOTP_TFTP_SERVERIP
-	bool "Enable BOOTP TFTP serverip"
-	depends on CMD_NET
-
-config BOOTP_PXE_CLIENTARCH
-	hex
-	depends on CMD_NET
-        default 0x16 if ARM64
-        default 0x15 if ARM
-        default 0 if X86
-
-config BOOTP_VCI_STRING
-	string
-	depends on CMD_NET
-	default "U-Boot.armv7" if CPU_V7 || CPU_V7M
-	default "U-Boot.armv8" if ARM64
-	default "U-Boot.arm" if ARM
-	default "U-Boot"
-
 endif   # if NET
diff --git a/net/eth_legacy.c b/net/eth_legacy.c
index a28032823e..41f5263526 100644
--- a/net/eth_legacy.c
+++ b/net/eth_legacy.c
@@ -239,12 +239,9 @@ int eth_initialize(void)
 {
 	int num_devices = 0;
 
-#ifndef CONFIG_BOARD_EARLY_INIT_R
 	eth_devices = NULL;
 	eth_current = NULL;
 	eth_common_init();
-#endif
-
 	/*
 	 * If board-specific initialization exists, call it.
 	 * If not, call a CPU-specific one
-- 
2.29.0

